src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -374,6 +391,150 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         BellStyle(currentStyle);
     }
 
+    // Method Description:
+    // - Construct _CurrentBellSounds by importing the _inherited_ value from the model
+    // - Adds a PropertyChanged handler to each BellSoundViewModel to propagate changes to the model
+    void ProfileViewModel::_InitializeCurrentBellSounds()
+    {
+        _CurrentBellSounds = winrt::single_threaded_observable_vector<Editor::BellSoundViewModel>();
+        if (const auto soundList = _profile.BellSound())
+        {
+            for (const auto&& bellSound : soundList)
+            {
+                auto vm = winrt::make<BellSoundViewModel>(bellSound);
+                vm.PropertyChanged({ this, &ProfileViewModel::_BellSoundVMPropertyChanged });
+                _CurrentBellSounds.Append(vm);
+            }
+        }
+        _MarkDuplicateBellSoundDirectories();
+        _NotifyChanges(L"CurrentBellSounds");
+    }
+
+    // Method Description:
+    // - If the current layer is inheriting the bell sound from its parent,
+    //   we need to copy the _inherited_ bell sound list to the current layer
+    //   so that we can then apply modifications to it
+    void ProfileViewModel::_PrepareModelForBellSoundModification()
+    {
+        if (const auto inheritedSounds = _profile.BellSound(); !_profile.HasBellSound() && inheritedSounds)
+        {
+            auto newSounds{ winrt::single_threaded_vector<winrt::hstring>() };
+            for (const auto sound : inheritedSounds)
+            {
+                newSounds.Append(sound);
+            }
+            _profile.BellSound(newSounds);

```

Comment:
Will this cause the inherited sounds to be written to the json for this profile even if no edits are made to them?

Reply:
Nope. From testing it just now:
- clicking save when no changes have been made --> bell sound isn't added
- clicking save after adding a sound --> sound gets serialized
- clicking save after adding a sound them deleting the added sound --> bell sound serialized as empty list

NOTE: I did find a bug where adding a sound when none are defined in the inheritance tree would cause a crash! That'll be fixed in the upcoming commit.